test/e2e/helpers.go

```diff
@@ -875,34 +875,40 @@ func getFlatcarVersions(ctx context.Context, location, galleryName, k8sVersion s
 func getFlatcarK8sVersions(ctx context.Context, location, communityGalleryName string) semver.Versions {
 	Logf("Finding Flatcar images and versions in community gallery %q in location %q", communityGalleryName, location)
 	var versions semver.Versions
-	k8sVersion := regexp.MustCompile(`flatcar-stable-amd64-capi-v(\d+)\.(\d+).(\d+)`)
-	communityGalleryImagesClient := newCommunityGalleryImagesClient()
-	communityGalleryImageVersionsClient := newCommunityGalleryImageVersionsClient()
-	var images []*armcompute.CommunityGalleryImage
-	pager := communityGalleryImagesClient.NewListPager(location, communityGalleryName, nil)
-	for pager.More() {
-		nextResult, err := pager.NextPage(ctx)
-		Expect(err).NotTo(HaveOccurred())
-		images = append(images, nextResult.Value...)
-	}
 
-	for _, image := range images {
-		var imageVersions []*armcompute.CommunityGalleryImageVersion
-		pager := communityGalleryImageVersionsClient.NewListPager(location, communityGalleryName, *image.Name, nil)
-		for pager.More() {
-			nextResult, err := pager.NextPage(ctx)
-			Expect(err).NotTo(HaveOccurred())
-			imageVersions = append(imageVersions, nextResult.Value...)
-		}
-
-		if len(imageVersions) == 0 {
-			continue
-		}
-
-		match := k8sVersion.FindStringSubmatch(*image.Name)
-		stringVer := fmt.Sprintf("%s.%s.%s", match[1], match[2], match[3])
-		versions = append(versions, semver.MustParse(stringVer))
-	}
+	// TODO: remove this once Flatcar image is patched https://github.com/kubernetes-sigs/image-builder/issues/1346
+	// Temporarily pin to v1.26.10
+	versions = append(versions, semver.MustParse("1.26.10"))
```

Comment:
you can do this by changing https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/config/azure-dev.yaml#L173 so it bypasses this code completely